src/parser/transform/statement/transform_create_table.cpp

```diff
@@ -95,6 +96,31 @@ unique_ptr<CreateStatement> Transformer::TransformCreateTable(duckdb_libpgquery:
 		case duckdb_libpgquery::T_PGColumnDef: {
 			auto cdef = (duckdb_libpgquery::PGColumnDef *)c->data.ptr_value;
 			auto centry = TransformColumnDefinition(cdef);
+			if (centry.type.IsSerial()) {
+				// create a sequence associated with the column
+				auto sequence = make_unique<CreateSequenceInfo>();
+				string seq_name = info->table + "_" + centry.name + "_seq";
+				sequence->name = seq_name;
+				if (centry.type.id() == LogicalTypeId::SMALLSERIAL) {
+					sequence->max_value = NumericLimits<int16_t>::Maximum();
+					centry.type = LogicalType(LogicalType::SMALLINT);
+				} else if (centry.type.id() == LogicalTypeId::SERIAL) {
+					sequence->max_value = NumericLimits<int32_t>::Maximum();
+					centry.type = LogicalType(LogicalType::INTEGER);
+				} else {
+					centry.type = LogicalType(LogicalType::BIGINT);
```

Comment:
note here that I don't keep the `SERIAL` type associated with the column post-table creation; this made everything else much easier to do (no need for e.g. casting values)

Reply:
Agreed using built-in types is the way to go here - but as mentioned before I would not add it as a `LogicalType` to begin with, but rather add it as either a constraint or an extra annotation to the column